### PR TITLE
Add optional nexus service

### PR DESCRIPTION
**Pull request by jojomomojo**
To use: add `COMPOSE_FILE=docker-compose.yml:compose-nexus.yml` to `.env`

Nexus is a package repo cache, using it as a local cache for docker, apt, pypi, npm, go, etc.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+nexus-password:
+	@docker-compose exec nexus cat /nexus-data/admin.password
+
+nexus-test:
+	@docker run -ti docker.defn.sh/ubuntu:focal bash -c "apt update; apt install -y ca-certificates; sed 's#http://archive.ubuntu.com/ubuntu#https://nexus.defn.sh/repository/ubuntu#; s#http://security.ubuntu.com/ubuntu#https://nexus.defn.sh/repository/ubuntu-security#' -i /etc/apt/sources.list; apt update; apt upgrade -y; apt install -y vim"
+	@docker run -ti docker.defn.sh/debian:buster bash -c "apt update; apt install -y ca-certificates; sed 's#http://deb.debian.org/debian#https://nexus.defn.sh/repository/debian#; s#http://security.debian.org/debian-security#https://nexus.defn.sh/repository/debian-security#' -i /etc/apt/sources.list; apt update; apt upgrade -y; apt install -y vim"
```
Comment (bellaweo) [inline Makefile:6]: should docker.defn.sh be dynamic and refer to the $KITT_DOMAIN var?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+go-env:
+	go env -w GO111MODULE=on
+	go env -w GOPROXY="https://nexus.defn.sh/repository/go/,direct"
```
Comment (bellaweo) [inline Makefile:10]: $KITT_DOMAIN here as well?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+nexus-password:
+	@docker-compose exec nexus cat /nexus-data/admin.password
+
+nexus-test:
+	@docker run -ti docker.defn.sh/ubuntu:focal bash -c "apt update; apt install -y ca-certificates; sed 's#http://archive.ubuntu.com/ubuntu#https://nexus.defn.sh/repository/ubuntu#; s#http://security.ubuntu.com/ubuntu#https://nexus.defn.sh/repository/ubuntu-security#' -i /etc/apt/sources.list; apt update; apt upgrade -y; apt install -y vim"
```
Comment (bellaweo) [inline Makefile:5]: should docker.defn.sh be dynamic and refer to the $KITT_DOMAIN var?

Comment (jojomomojo) [reply Makefile]: I've changed defn.sh to KITT_DOMAIN, except for .drone.yml.  Solve that separately.

File: Makefile (hunk 2 above)
Comment (jojomomojo) [reply Makefile]: I've replaced defn.sh with KITT_DOMAIN.

**Comment by jojomomojo** [comment]
I assume the most basic use case: that KITT_DOMAIN is set in the environment.  For projects outside of kitt, that use kitt, the .env could come from consul/vault.  Pass is something I'd like to keep just for kitt to bootstrap it until consul/vault can be used.

**Pull request merged**